databuilder/tables/beta/tpp.py

```diff
@@ -87,6 +105,46 @@ class clinical_events(EventFrame):
     ctv3_code = Series(CTV3Code)
     numeric_value = Series(float)
 
+    def with_codes(
+        self,
+        codelist,
+        after=None,
+        before=None,
+        on_or_after=None,
+        on_or_before=None,
+        between=None,
+        on_or_between=None,
+    ):
+        events = self
+
+        # TODO: We should make `Codelist` iterable so we don't need this conditional
+        codes = codelist.codes if hasattr(codelist, "codes") else codelist
```

Comment:
Sounds like `Codelist` should have a `.system` attribute.

Reply:
I did think about this, and there may be other good reasons for having that attribute. But in this specific context, given that we still want to handle plain old lists of codes, I don't think it would help us.